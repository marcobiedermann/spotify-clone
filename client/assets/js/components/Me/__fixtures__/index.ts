const image = {
  height: 100,
  url: '',
  width: 100,
};

const me = {
  display_name: '',
  id: '',
  images: [image],
};

export default me;
