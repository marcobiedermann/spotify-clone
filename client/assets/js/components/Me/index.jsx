import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import styles from './style.css';

const Me = ({
  display_name,
  id,
  images,
}) => (
  <div className={styles.me}>
    <figure className={styles.me__image}>
      {images[0] && (
        <img src={images[0].url} alt={display_name} width="32" height="32" />
      )}
    </figure>
    <Link to={`/users/${id}`}>
      {display_name}
    </Link>
  </div>
);

Me.propTypes = {
  display_name: PropTypes.string,
  id: PropTypes.string,
  images: PropTypes.arrayOf(PropTypes.shape()),
};

Me.defaultProps = {
  display_name: '',
  id: '',
  images: [],
};

export default Me;
