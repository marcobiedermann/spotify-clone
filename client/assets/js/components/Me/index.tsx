import classNames from 'classnames';
import React, { FC } from 'react';
import { Link } from 'react-router-dom';
import Image, { ImageProps } from '../Image';
import styles from './style.css';

export interface MeProps {
  className?: string;
  display_name: string;
  id: string;
  images: ImageProps[];
}

const Me: FC<MeProps> = (props) => {
  const { className, display_name, id, images } = props;

  return (
    <div className={classNames(className, styles.me)}>
      <figure className={styles.me__image}>
        {images[0] && <Image url={images[0].url} alt={display_name} width={32} height={32} />}
      </figure>
      <Link to={`/users/${id}`}>{display_name}</Link>
    </div>
  );
};

export default Me;
