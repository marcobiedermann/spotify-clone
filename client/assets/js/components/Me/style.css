.me {
  align-items: center;
  display: flex;
  flex-wrap: wrap;
}

.me__image {
  margin-right: 1em;
}

.me__image img {
  border-radius: 50%;
}
