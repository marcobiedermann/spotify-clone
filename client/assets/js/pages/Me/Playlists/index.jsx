import React from 'react';
import { Helmet } from 'react-helmet';

const PlaylistsPage = () => (
  <div>
    <Helmet>
      <title>Playlists</title>
    </Helmet>
    Playlists
  </div>
);

export default PlaylistsPage;
