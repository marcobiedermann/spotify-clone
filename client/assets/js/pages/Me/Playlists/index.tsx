import React, { FC } from 'react';
import { Helmet } from 'react-helmet';
import { RouteChildrenProps } from 'react-router-dom';

const PlaylistsPage: FC<RouteChildrenProps> = () => (
  <>
    <Helmet>
      <title>Playlists</title>
    </Helmet>
    Playlists
  </>
);

export default PlaylistsPage;
