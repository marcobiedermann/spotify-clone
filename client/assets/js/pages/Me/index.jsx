import PropTypes from 'prop-types';
import React from 'react';
import {
  Route,
  Switch,
} from 'react-router-dom';
import PlaylistsPage from './Playlists';

const MePage = ({ match }) => (
  <Switch>
    <Route
      path={`${match.url}/playlists`}
      component={PlaylistsPage}
    />
    <Route
      path={match.url}
      component={() => (
        <div>
          Me
        </div>
      )}
    />
  </Switch>
);

MePage.propTypes = {
  match: PropTypes.shape({
    url: PropTypes.string,
  }),
};

MePage.defaultProps = {
  match: {
    url: '',
  },
};

export default MePage;
