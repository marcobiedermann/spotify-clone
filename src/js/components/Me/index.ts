import Me from './Me';

export default Me;
export * from './Me';
