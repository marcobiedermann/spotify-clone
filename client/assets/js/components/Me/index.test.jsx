import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Me from '.';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Me />
      </MemoryRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
