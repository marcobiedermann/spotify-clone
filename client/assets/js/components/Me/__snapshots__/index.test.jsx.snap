// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="me"
>
  <figure
    className="me__image"
  />
  <a
    href="/users/"
    onClick={[Function]}
  >
    
  </a>
</div>
`;
